lmfdb/characters/test_characters.py

```diff
@@ -116,11 +106,11 @@ def test_dirichletcharbig(self):
             This test also makes sure the code scales a little bit.
         """
         W = self.tc.get('/Character/Dirichlet/40487/5')
-        assert '40486' in W.get_data(as_text=True)
-        assert '12409' in W.get_data(as_text=True), "log on generator"
```

Comment:
@AndrewVSutherland Since we're no longer using `WebDirichletCharacter` for modulus between 10,000 and 100,000, the `Values on generators` doesn't appear on this page, so this test failed. I therefore changed it to check the field of values; but do let me know if I misunderstood something, or if the behaviour should be otherwise.

Reply:
Actually, I realized that we can keep the values on generators and the values, because Pari now supports Conrey characters and can compute these very efficiently.  To compute the value of chi_q(m,n) you can use:

```
 def conreyangle(q,m,n):
     return Pari("G=znstar(%d,1); chareval(G,znconreylog(G,%d),%d)"%(q,m,n))
```

which will return a rational number r such that chi_q(m,n) = exp(2*pi*r).  This is very fast even for 10 or 20 digit values of q.

To get generators you can use `Integers(q).unit_gens()`.

With this we can easily keep the values on generators and values sections for all Dirichlet characters, as well as the ability to compute values, just as we do for characters in the database.

Reply:
OK thanks for the feedback @AndrewVSutherland , I'll aim to get `Values on generators` as well as `First values` back onto the pages for large modulus via Pari.